pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -136,6 +136,10 @@ func (r *ReconcileClusterDeployment) Reconcile(request reconcile.Request) (recon
 	cdLog.Info("reconciling cluster deployment")
 	cd = cd.DeepCopy()
 
+	if cd.Status.ClusterUUID == "" {
```

Comment:
I'm wondering if ClusterUUID really belongs in Status. Is there ever a case when the end-user wants to specify a ClusterUUID? (to adopt an existing cluster, for example)

Reply:
If it were moved to spec, is it ok for us to automatically fill that out if unset (which it normally would be?) I guess that would probably be fine.